cmd/ebrelayer/relayer/ethereum.go

```diff
@@ -228,6 +231,11 @@ func (sub EthereumSub) Start(completionEvent *sync.WaitGroup) {
 			// The user who starts this must provide a valid last processed block
 			if lastProcessedBlock.Cmp(big.NewInt(0)) == 0 {
 				lastProcessedBlock.Sub(endingBlock, big.NewInt(1))
+			} else if catchUpNeeded {
+				// if we need to catch up, then do that and let the relayer know that we don't need to catch up again
+				catchUpNeeded = false
```

Comment:
If catchUpNeeded, then just set is as false, how about the lastProcessedBlock, no update? If use the same block number, the all events in this block number will be processed again.

Reply:
@ElliotFriedman

Reply:
You set catchUpNeeded to false, remember the last block number for that run, then, the next time you see a block, you just grab data from 50 blocks ago.

Reply:
I have tested it with thousands of transactions locally and it works just fine.